srml/bridge/src/lib.rs

```diff
@@ -160,6 +161,39 @@ impl<T: Trait> Module<T> {
 	}
 }
 
+// A naive way to check whether a `child` header is an ancestor
+// of an `ancestor` header. For this it requires a proof which
+// is a header chain, This could be updated to use something like
+// Log2 Ancestors (#2053) in the future.
+fn verify_ancestry<H>(proof: Vec<H>, ancestor: H, child: H) -> std::result::Result<(), Error>
+where
+	H: Header
+{
+	let mut curr_header = &proof[0];
+	if curr_header.hash() != child.hash() {
+		return Err(Error::AncestorNotFound);
+	}
+
+	let mut parent_hash = curr_header.parent_hash();
+
+	// If we find that the header's parent hash matches our ancestor's hash we're done
+	for i in 1..proof.len() {
```

Comment:
Iterate over the headers directly, not the indices.

`for header in &proof[1..].iter()`